src/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/NativeLayoutInfoLoadContext.cs

```diff
@@ -132,11 +132,17 @@ internal TypeDesc GetType(ref NativeParser parser)
                         // Skip encoded bounds and lobounds
                         uint boundsCount = parser.GetUnsigned();
                         while (boundsCount > 0)
+                        {
                             parser.GetUnsigned();
+                            boundsCount--;
```

Comment:
We don't have test coverage for the type loader and multidimensional arrays on the TFS side that would catch this?